vermouth/processors/make_bonds.py

```diff
@@ -76,25 +80,36 @@ def bonds_from_distance(system, fudge=1.1):
         certain distance from each other. It is probably disconnected.
     """
     system = nx.compose_all(system.molecules)
-    idx_to_nodenum = {idx: n for idx, n in enumerate(system)}
+    idx_to_nodenum = {
+        idx: n
+        for idx, n in enumerate(
+                subn
+                for subn in system
+                if system.nodes[subn].get('element') in VDW_RADII
+        )
+    }
     max_dist = max(VDW_RADII.get(node.get('element'), 0.2) for node in system.nodes.values())
-    positions = np.array([system.node[n]['position'] for n in system], dtype=float)
+    positions = np.array([
+        node['position']
+        for node in system.nodes.values()
+        if node.get('element') in VDW_RADII
```

Comment:
Get them from `idx_to_nodenum`

Reply:
From my quick test, it makes the code a bit less readable. It means `idx_to_nodenum` must be an `OrderedDict` because of python < 3.6; and it leads to many more indirections to access the elements.

Reply:
Ok, good point about the ordereddict.